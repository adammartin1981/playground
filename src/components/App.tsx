import * as React from 'react'

const GiftsDisplay = ({ gifts }: { gifts: string[] }) => (
  <>
    {gifts.map((gift) => {
      return <div data-testid="gift">Gift</div>
    })}
  </>
)

export const App = () => {
  const [gifts, setGifts] = React.useState<string[]>([])

  const addGifts = () => setGifts([...gifts, 'New Gift'])

  return (
    <div className="app">
      <div>Hello</div>
      <GiftsDisplay gifts={gifts} />

      <button data-testid="addGift" onClick={addGifts}>
        Add Gifts
      </button>
    </div>
  )
}
