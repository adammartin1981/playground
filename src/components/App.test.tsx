import { App } from './App'
import * as React from 'react'

import { shallow, configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe('App', () => {
  const app = shallow(<App />)
  it('should sort the App Component', () => {
    expect(shallow(<App />).is('.foo')).toBe(true)
  })

  it('should do snapshots', () => {
    expect(app).toMatchSnapshot()
  })
})
