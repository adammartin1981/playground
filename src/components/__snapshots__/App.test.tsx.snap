// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should do snapshots 1`] = `
<div
  className="foo"
/>
`;

exports[`App should do wallaby snapshot 1`] = `
<div
  className="foo"
/>
`;

exports[`App should do wallaby snapshot 2`] = `
<div
  className="foo"
/>
`;
